miio/airqualitymonitor_miot.py

```diff
@@ -0,0 +1,254 @@
+import enum
+import logging
+
+import click
+
+from .click_common import command, format_output
+from .exceptions import DeviceException
+from .miot_device import MiotDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_AIRQUALITYMONITOR_CGDN1 = "cgllc.airm.cgdn1"
+
+_MAPPING_CGDN1 = {
+    # Source https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:air-monitor:0000A008:cgllc-cgdn1:1
+    # Environment
+    "humidity": {"siid": 3, "piid": 1},  # [0, 100] step 1
+    "pm25": {"siid": 3, "piid": 4},  # [0, 1000] step 1
+    "pm10": {"siid": 3, "piid": 5},  # [0, 1000] step 1
+    "temperature": {"siid": 3, "piid": 7},  # [-30, 100] step 0.00001
+    "co2": {"siid": 3, "piid": 8},  # [0, 9999] step 1
+    # Battery
+    "battery": {"siid": 4, "piid": 1},  # [0, 100] step 1
+    "charging_state": {
+        "siid": 4,
+        "piid": 2,
+    },  # 1 - Charging, 2 - Not charging, 3 - Not chargeable
+    "voltage": {"siid": 4, "piid": 3},  # [0, 65535] step 1
+    # Settings
+    "start_time": {"siid": 9, "piid": 2},  # [0, 2147483647] step 1
+    "end_time": {"siid": 9, "piid": 3},  # [0, 2147483647] step 1
+    "monitoring_frequency": {"siid": 9, "piid": 4},  # 1, 60, 300, 600, 0
+    "screen_off": {"siid": 9, "piid": 5},  # 15, 30, 60, 300, 0
+    "device_off": {"siid": 9, "piid": 6},  # 15, 30, 60, 0
+    "temperature_unit": {"siid": 9, "piid": 7},
+}
+
+
+class AirQualityMonitorMiotException(DeviceException):
+    pass
+
+
+class ChargingStateCGDN1(enum.Enum):
+    Unplugged = 0  # Not mentioned in the spec
+    Charging = 1
+    NotCharging = 2
+    NotChargable = 3
+
+
+class MonitoringFrequencyCGDN1(enum.Enum):
+    Every1Second = 1
+    Every1Minute = 60
+    Every5Minutes = 300
+    Every10Minutes = 600
+    NotSet = 0
+
+
+class ScreenOffCGDN1(enum.Enum):
+    After15Seconds = 15
+    After30Seconds = 30
+    After1Minute = 60
+    After5Minutes = 300
+    Never = 0
+
+
+class DeviceOffCGDN1(enum.Enum):
+    After15Minutes = 15
+    After30Minutes = 30
+    After1Hour = 60
+    Never = 0
+
+
+class DisplayTemeratureUnitCGDN1(enum.Enum):
+    Celcius = "c"
+    Fahrenheit = "f"
+
+
+class AirQualityMonitorCGDN1Status:
+    """Container of air quality monitor CGDN1 status."""
+
+    def __init__(self, data):
+        self.data = data
```

Comment:
Add an example `data` content to the docstring, that can be helpful for anyone looking at the code without having access to such device.

Reply:
Done